lib/computed_model.rb

```diff
@@ -237,7 +239,12 @@ def bulk_load_and_compute(objs, deps, **options)
             obj.send(:"compute_#{dep_name}")
           end
         elsif @__computed_model_loaders.key?(dep_name)
-          @__computed_model_loaders[dep_name].call(objs, plan.subdeps_hash[dep_name], **options)
+          l = @__computed_model_loaders[dep_name]
+          keys = objs.map { |o| o.instance_exec(&(l.key_proc)) }
+          subobj_by_key = l.load_proc.call(keys, plan.subdeps_hash[dep_name], **options)
+          objs.each.with_index do |obj, i|
+            obj.send(:"#{dep_name}=", subobj_by_key[keys[i]])
+          end
```

Comment:
This can be zip

```suggestion
          objs.zip(keys) do |obj, key|
            obj.send(:"#{dep_name}=", subobj_by_key[key])
          end
```